lib/std/priority_queue.zig

```diff
@@ -204,8 +204,13 @@ pub fn PriorityQueue(comptime T: type) type {
         pub fn update(self: *Self, elem: T, new_elem: T) !void {
             var update_index: usize = linearSearch(elem, self.items);
             assert (update_index >= 0 and update_index < self.items.len);
-            _ = self.removeIndex(update_index);
-            try self.add(new_elem);
+            // Heapreplace:
```

Comment:
* No trivial comments please
* Deciding whether this is an increase/decrease depends on the value at `update_index` and `new_elem` (compared using `compareFn`)
* You need a `siftUp`, factor out the implementation in `addUnchecked` and use it in both places
* The use of `std.mem.swap` is a nice touch :sunglasses:

Reply:
> No trivial comments please 

yes, will remove those

> Deciding whether this is an increase/decrease depends on the value at update_index and new_elem (compared using compareFn)

> You need a siftUp, factor out the implementation in addUnchecked and use it in both places 

Hmm, okay, will do sift up or down based on compareFn

Reply:
Please do review my current implementation, I've refactored with all of the above suggestions